File: WebApplication1/WebApplication1/Models/Vozilo.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace WebApplication1.Models
{
    public class Vozilo
    {
        [Key]
        public int Id { get; set; }

        public string Naziv { get; set; }

        public string Marka { get; set; }

        public string Model { get; set; }

        public int GodinaProizvodnje { get; set; }

        public int BrojSedista { get; set; }

        public string TipVozila { get; set; }

        public int RentACarServisID { get; set; }

        public int FilijalaID { get; set; }

        public double Ocena { get; set; }

        [NotMapped]
        public List<DateTime> ZauzetiDatumi { get; set; }

        public string ZauzetiDatumiString { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }

        public void PretvoriUJson()
        {
            string output = JsonConvert.SerializeObject(ZauzetiDatumi);
            ZauzetiDatumiString = output;
        }

        public void PretvoriUListu()
        {
            if (ZauzetiDatumiString != null)
            {
                List<DateTime> deserializedProduct = JsonConvert.DeserializeObject<List<DateTime>>(ZauzetiDatumiString);
                ZauzetiDatumi = deserializedProduct;
            }
            else
            {
                ZauzetiDatumi = new List<DateTime>();
            }
        }
    }
}

File: projekat/src/app/entities/vozilo/vozilo.ts
import { NumberValueAccessor } from '@angular/forms';
import { RentACarServis } from '../rentacar/rentacar';

export class Vozilo {
    id: number;
    naziv: string;
    marka: string;
    model: string;
    godinaProizvodnje: number;
    brojSedista: number;
    tipVozila: string;
    rentACarId:number;
    filijalaId:number;
    rowVersion:any[];
    ocena:number;


    constructor(id:number, naziv: string, marka: string, model: string, godinaProizvodnje: number, brojSedista: number, tipVozila: string, rowVersion:any[]) {
        this.id = id;
        this.naziv = naziv;
        this.marka = marka;
        this.model = model;
        this.godinaProizvodnje = godinaProizvodnje;
        this.brojSedista = brojSedista;
        this.tipVozila = tipVozila;
        this.rowVersion = rowVersion;
    }


}